gdal/ogr/ogrsf_frmts/geojson/libjson/json_object.c

```diff
@@ -10,6 +10,10 @@
  *
  */
 
+// For uselocale
```

Comment:
is this change really needed for this pull request ? I don't see the connection. I'd prefer to see it go through another one if possible
In ogr/ogrsf_frmts/geojson/libjson/GNUmakefile, we already have:
# -D_XOPEN_SOURCE=500 to enable strdup() definition in C11 mode
CPPFLAGS := -D_XOPEN_SOURCE=500 $(CPPFLAGS)

Reply:
At this point I believe it is necessary. But I don't know why. Last night I was seeing a "strdup undefined" error on OSX, with ld suggesting we #include <string.h>. But json_object.c *does* #include <string.h>. The problem *appears* to be that string.h on OSX defines strdup() only if _XOPEN_SOURCE >= 700L. And by default autoconf sets _XOPEN_SOURCE = 500L. So OSX strdup is undefined. Two obvious questions:
1. Why did it fail last night, and not before? Did my (seemingly) unrelated changes to configure.ac trigger an _XOPEN_SOURCE change? git should know; I'll have to ask. 
2. If _XOPEN_SOURCE = 700L is indeed required, would it not be better to set it once in configure.ac, rather than in json_object.c, json_tokener.c etc.?
Ref: https://lists.gnupg.org/pipermail/gnupg-devel/2017-February/032550.html

Is there a way to tag this PR a WIP (Work In Progress)? I'm still working on my new STRIP_SYSTEM_LIBRARY_PATHS macro, and it's likely it might pass all COIN build tests before we're both satisfied with it, and I wouldn't want to see it prematurely merged. Thanks!

Reply:
Okay. So configure sets #define _XOPEN_SOURCE 700; and ogr/ogrsf_frmts/geojson/libjson/GNUmakefile sets CPPFLAGS := -D_XOPEN_SOURCE=500 $(CPPFLAGS)

If you would like, I can try removing this line from this GNUmakefile, and the _XOPEN_SOURCE #definitions from json_object.c, json_tokener,c and see how it goes. 

I agree it should be a separate issue and PR, but we need to get the present #2395 resolved as well, and the OSX strdup() *appears* to block it. I might be able to create another branch for the _XOPEN_SOURCE issue, fix it and test locally, and push to you before continuing with the present #2395. Depends on how easily I can subvert the proj shadow problem. I've recently installed Fedora-32 beta, and it's system proj is 6.3.1; I could work from there (later).